lib/ipmi_quantaoem.c

```diff
@@ -164,7 +180,7 @@ oem_qct_get_evt_desc(struct ipmi_intf *intf, struct sel_event_record *rec)
 		}
 		/* check the platform type */
 		platform_id = oem_qct_get_platform_id(intf);
-		if (OEM_QCT_PLATFORM_PURLEY == platform_id) {
+		if ((OEM_QCT_PLATFORM_PURLEY == platform_id) || (OEM_QCT_PLATFORM_GRANTLEY == platform_id)) {
```

Comment:
I guess it's time to convert this `if` to a `switch`.
This will look much more readable and will also let you drop the `platform_id` variable:
```
switch(oem_qct_get_platform_id(intf)) {
case OEM_QCT_PLATFORM_PURLEY:
case OEM_QCT_PLATFORM_GRANTLEY:
    snprintf(desc, ...
    break;
default:
    break;
}
```